process_data.py

```diff
@@ -62,21 +63,37 @@ def get_parameters():
     return args
 
 
-def compute_snr(file_data1, file_data2, file_mask):
+def compute_snr_diff(file_data1, file_data2, file_mask):
     """
     Compute SNR based on two input data and a mask
     :param file_data1: image 1
     :param file_data1: image 2
     :param file_mask: mask where to compute SNR
     :return: float: SNR rounded at 2 decimals
     """
-    print("Compute SNR...")
+    print("Compute SNR_diff...")
     sct.run("sct_image -i " + file_data1 + "," + file_data2 + " -concat t -o data_concat.nii.gz")
     status, output = sct.run("sct_compute_snr -i data_concat.nii.gz -vol 0,1 -m " + file_mask)
     # parse SNR info
-    snr = np.float(output[output.index("SNR_diff =") + 11:])
-    return round(snr, 2)  # round at 2 decimals
+    snr_diff = np.float(output[output.index("SNR_diff =") + 11:])
+    return round(snr_diff, 2) # round at 2 decimals
 
+def compute_snr_single(file_data, file_mask):
+    """
+    Compute SNR based on a single image and a mask
+    :param file_data: image
+    :param file_mask: mask for ROI_body
+    :return: float: SNR rounded at 2 decimals
+    """
+    print("Compute SNR_single...")
+    # Convert image to array
+    data = Image(file_data).data
+    # Convert mask to array
+    wm_mask = Image(file_mask).data
+    roi= np.where(wm_mask == 1)
+    roi_data = data[roi]
+    snr_single = np.mean(roi_data) / np.std(roi_data)
```

Comment:
SNR should be computed slice wise

Reply:
@jcohenadad yes, that's right. Adjusting it now.